s3_file_field/testing.py

```diff
@@ -0,0 +1,106 @@
+
+
+class S3FileFieldTestClient:
+    def __init__(self, api_client, base_url: str):
+        """
+        Initialize a S3FileField test client.
+
+        Args:
+            api_client: An instance of `rest_framework.test.APIClient`.
+            base_url: The relative path that s3_file_field.urls is mounted at.
+        """
+        self.api_client = api_client
+        self.base_url = base_url.rstrip('/')
+
+    def _initialize_upload(self, file: _File, field_id: str) -> Dict:
+        resp = self.api_client.post(
+            f'{self.base_url}/upload-initialize/',
+            {
+                'field_id': field_id,
+                'file_name': file.name,
+                'file_size': file.size,
+            },
+            format='json',
+        )
+
+        assert resp.status_code < 400
+        return resp.json()
+
+    def _upload_part(self, part_bytes: bytes, part_initialization: Dict):
+        resp = requests.put(part_initialization['upload_url'], data=part_bytes)
+        resp.raise_for_status()
+
+        etag = resp.headers['ETag']
+
+        return {
+            'part_number': part_initialization['part_number'],
+            'size': part_initialization['size'],
+            'etag': etag,
+        }
+
+    def _upload_parts(self, file: _File, part_initializations: List[Dict]) -> List[Dict]:
+        return [
+            self._upload_part(file.stream.read(part_initialization['size']), part_initialization)
+            for part_initialization in part_initializations
+        ]
+
+    def _complete_upload(self, multipart_info: Dict, upload_infos: List[Dict]) -> None:
+        resp = self.api_client.post(
+            f'{self.base_url}/upload-complete/',
+            {
+                'upload_id': multipart_info['upload_id'],
+                'parts': upload_infos,
+                'upload_signature': multipart_info['upload_signature'],
+            },
+            format='json',
+        )
+
+        assert resp.status_code < 400
+        completion_data = resp.json()
+
+        complete_resp = requests.post(completion_data['complete_url'], data=completion_data['body'])
+        complete_resp.raise_for_status()
+
+    def _finalize(self, multipart_info: Dict) -> str:
+        resp = self.api_client.post(
+            f'{self.base_url}/finalize/',
+            {
+                'upload_signature': multipart_info['upload_signature'],
+            },
+            format='json',
+        )
+        assert resp.status_code < 400
+        return resp.json()['field_value']
+
+    def upload_file(self, file_stream: BinaryIO, file_name: str, field_id: str) -> str:
+        file = _File.from_stream(file_stream, file_name)
+        multipart_info = self._initialize_upload(file, field_id)
+        upload_infos = self._upload_parts(file, multipart_info['parts'])
+        self._complete_upload(multipart_info, upload_infos)
+        field_value = self._finalize(multipart_info)
+        return field_value
```

Comment:
I can see how not wrapping this in the dict makes things simpler, but I think it's smart to maintain parity with the python client. It's possible that tests will use this as a drop in replacement for the python client, which would break stuff.

Reply:
Do you think it makes sense to maintain parity by updating the python client instead? I guess I'm just confused what the purpose of wrapping the field value in a dict is. Is it a way of implicit documentation? Regardless of which direction we go I do agree we should maintain parity with the python client.

Reply:
Hypothetically we might want to add more info to the return value in the future, so using a dict future proofs it.

I think fixing it here makes a lot more sense for now, as we wouldn't have to force downstream clients to bump their version. Changing the API surface should be it's own issue/PR.

Reply:
Sounds good, I'll roll back that change